src/H5VLcallback.c

```diff
@@ -1437,6 +1486,50 @@ H5VLattr_get(void *obj, hid_t connector_id, H5VL_attr_get_t get_type, hid_t dxpl
     FUNC_LEAVE_API_NOINIT(ret_value)
 } /* end H5VLattr_get() */
 
+/*-------------------------------------------------------------------------
+ * Function:    H5VLattr_get_vararg
+ *
+ * Purpose:     Gets information about the attribute
+ *
+ * Note:        Same as H5VLattr_get, but uses varargs instead of a va_list
+ *
+ * Return:      Success:    Non-negative
+ *              Failure:    Negative
+ *
+ *-------------------------------------------------------------------------
+ */
+herr_t
+H5VLattr_get_vararg(void *obj, hid_t connector_id, H5VL_attr_get_t get_type, hid_t dxpl_id,
+                    void **req /*out*/, ...)
```

Comment:
I may very well have missed the discussions surrounding adding var_args versions of the VOL callbacks, but did we end up coming to a conclusion on these? While it looks like it did help clean up the passthrough connector a bit, it seems very inconvenient to have to add (and maintain) separate versions of each VOL call. Not to mention that it seems like a connector author may have problems determining when one is needed vs. the other.

Reply:
Yes, we discussed adding them, a while ago, so they are in.   Having two versions is overkill to me also, but the va_list version was "legacy" at the time I added the vararg version.   If we're happy with it enough to justify [another] breaking change to the VOL connector callbacks, I can remove the va_list version.

Reply:
Ok, I think we should definitely discuss this with everyone then to see what the consensus is. IMO, it's still a good time to finalize any breaking changes, and, IMO, having two versions will probably just end up being a nightmare for both maintenance and cognitive overhead in the long run.

Reply:
> Ok, I think we should definitely discuss this with everyone then to see what the consensus is. IMO, it's still a good time to finalize any breaking changes, and, again IMO, having two versions will probably just end up being a nightmare for both maintenance and cognitive overhead in the long run.

Agree - I'll add it to the agenda for next Monday's technical meeting

Reply:
Elena is open to changing the existing functions to use varargs